soroban-env-host/src/storage.rs

```diff
@@ -312,19 +315,90 @@ impl Storage {
         }
     }
 
-    /// Marks the key as a read if we expect the host consumer to read this key.
+    /// Bumps `key` to live for at least `bump_by_ledgers` from now (not
+    /// counting the current ledger).
     ///
-    /// In [FootprintMode::Recording] mode, records the access.
+    /// This operation is only defined within a host as it relies on ledger
+    /// state.
     ///
-    /// In [FootprintMode::Enforcing] mode, succeeds only if the access has been
-    /// declared in the [Footprint].
-    pub fn touch_key(&mut self, key: &Rc<LedgerKey>, budget: &Budget) -> Result<(), HostError> {
-        let _span = tracy_span!("touch key");
-        let ty = AccessType::ReadOnly;
-        match self.mode {
-            FootprintMode::Recording(_) => self.footprint.record_access(key, ty, budget),
-            FootprintMode::Enforcing => self.footprint.enforce_access(key, ty, budget),
+    /// This is always considered a 'read-only' operation, even though it might
+    /// modify the internal storage state.
```

Comment:
Maybe extend this comment: "The only fields of the "read-only" entries it modifies are their expiry ledgers, and the higher level logic that processes the side effects of a transaction will separately extract, reconcile (by taking a maximum over all updates to a given key) and write back to stable storage _just_ the changed expiry times of "read-only" entries, not even considering the possibility of anything else about them changing."

Reply:
Updated the comment with small edits (`bump` correctly maintains max expiration and the potential process of reconciling RO bumps in core seems way too high level and specific for this doc)